### PR TITLE
MIBM-480_V2: adding navigator to controllers

### PR DESCRIPTION
**Pull request by PasGat**

### CODE REVIEW

File: app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala [hunk 1]
```diff
@@ -35,12 +36,13 @@ class EoriNumberController @Inject()(
...
   view: EoriNumberView,
-  mibConnector: MibConnector
+  mibConnector: MibConnector,
+  navigator: Navigator
 )(implicit ec: ExecutionContext, appConfig: AppConfig)
     extends DeclarationJourneyUpdateController with Logging {
 
   private def backButtonUrl(implicit request: DeclarationJourneyRequest[_]) =
-    backToCheckYourAnswersIfCompleteElse(routes.CustomsAgentController.onPageLoad())
+    backToCheckYourAnswersIfCompleteElse(CustomsAgentController.onPageLoad())
```
Comment (jordanrowe) [inline app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala:45]: potential to extend the navigator to have .previousPage() as well

**Review by jordanrowe — APPROVED** [review]

File: app/uk/gov/hmrc/merchandiseinbaggage/controllers/Navigator.scala [hunk 2]
```diff
@@ -40,11 +41,13 @@ object Navigator {
...
   )
 
-  def nextPageWithAnswer: Map[String, YesNo => Call] = Map(
-    CustomsAgentController.onPageLoad().url -> customsAgent
+  def nextPageWithAnswer[T]: Map[String, T => Call] = Map(
+    GoodsDestinationController.onPageLoad().url -> goodsDestination,
+    CustomsAgentController.onPageLoad().url     -> customsAgent
   )
 
   def nextPageWithIndex: Map[String, (YesNo, JourneyType, Int) => Call] = Map(
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/merchandiseinbaggage/controllers/Navigator.scala:53]: all the above defs could be vals or lazy vals ?

**Review by sureshhmrc — APPROVED** [review]

File: app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala (hunk 1 above)
Comment (PasGat) [reply app/uk/gov/hmrc/merchandiseinbaggage/controllers/EoriNumberController.scala:45]: sure, i was going to do as part of second iteration with backlink too

File: app/uk/gov/hmrc/merchandiseinbaggage/controllers/Navigator.scala (hunk 2 above)
Comment (PasGat) [reply app/uk/gov/hmrc/merchandiseinbaggage/controllers/Navigator.scala:53]: sure, done

**Pull request merged**